tracee/tracee_test.go

```diff
@@ -0,0 +1,201 @@
+package tracee
+
+import (
+	"bytes"
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestReadArgFromBuff(t *testing.T) {
+	testCases := []struct {
+		name          string
+		input         []byte
+		expectedArg   interface{}
+		expectedError error
+	}{
+		{
+			name: "INT_T",
+			input: []byte{1, //INT_T
+				0xFF, 0xFF, 0xFF, 0xFF, //-1
+			},
+			expectedArg: int32(-1),
+		},
+		{
+			name: "UINT_T",
+			input: []byte{2, //UINT_T
+				0xFF, 0xFF, 0xFF, 0xFF, //4294967295
+			},
+			expectedArg: uint32(4294967295),
+		},
+		{
+			name: "LONG_T",
+			input: []byte{3, //LONG_T
+				0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, //-1
+			},
+			expectedArg: int64(-1),
+		},
+		{
+			name: "ULONG_T",
+			input: []byte{4, //ULONG_T
+				0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, //18446744073709551615
+			},
+			expectedArg: uint64(18446744073709551615),
+		},
+		{
+			name: "OFF_T_T",
+			input: []byte{5, //OFF_T_T
+				0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, //18446744073709551615
+			},
+			expectedArg: uint64(18446744073709551615),
```

Comment:
The first byte determines the type, correct? So if that's the case why is there only a single case for them all here: https://github.com/aquasecurity/tracee/blob/fc957f7d2b03157ce57ee473be2ef4122723eef4/tracee/tracee.go#L543